src/shogun/regression/LinearRidgeRegression.cpp

```diff
@@ -57,32 +83,26 @@ bool CLinearRidgeRegression::train_machine(CFeatures* data)
     	"Expected Real Features (%d) but got (%d)\n",
     	F_DREAL, data->get_feature_type());
 
-    CDenseFeatures<float64_t>* feats=(CDenseFeatures<float64_t>*) data;
-    int32_t num_feat=feats->get_num_features();
-
-    SGMatrix<float64_t> kernel_matrix(num_feat,num_feat);
-    SGMatrix<float64_t> feats_matrix(feats->get_feature_matrix());
-    SGVector<float64_t> y(num_feat);
-    SGVector<float64_t> tau_vector(num_feat);
-
-    tau_vector.zero();
-    tau_vector.add(m_tau);
-
-	linalg::matrix_prod(feats_matrix, feats_matrix, kernel_matrix, false, true);
-	linalg::add_diag(kernel_matrix, tau_vector);
-
-	auto labels = regression_labels(m_labels);
-	auto lab = regression_labels(m_labels)->get_labels();
-	linalg::matrix_prod(feats_matrix, lab, y);
-
-	auto decomposition = linalg::cholesky_factor(kernel_matrix);
-	y = linalg::cholesky_solver(decomposition, y);
-	set_w(y);
-
-	auto x_bar = linalg::colwise_sum(feats_matrix);
-	linalg::scale(x_bar, x_bar, 1.0 / ((float64_t)x_bar.size()));
-	auto intercept = linalg::mean(lab) - linalg::dot(lab, x_bar);
-	set_bias(intercept);
+	auto feats = data->as<CDenseFeatures<float64_t>>();
+	auto y = regression_labels(m_labels)->get_labels();
+	auto N = feats->get_num_vectors();
+	auto D = feats->get_num_features();
+
+	SGVector<float64_t> w;
```

Comment:
So the model contains both w and b and what we are doing is first find w, ignoring that b exists, and then find b provided the found w.

Is this standard practice in linear regression? I am a bit surprised that the equations to find w do not depend on the b since the gradient of the loss wrt to w has b.

Reply:
Ok. My question was probably wrong. Even if b does not appear explicitly in the expression for w, that does not mean it is not in there. It has just been substituted.

However, then the expression for w should be in principle different for the cases (1) w/ bias; (2) w/o bias. No?

Reply:
You are right! the expression for w was wrong, it assumed no bias. I am fixing it right now